packages/x-components/src/components/__tests__/base-grid.spec.ts

```diff
@@ -68,30 +76,48 @@ describe('testing Base Grid', () => {
              <p data-test="default-slot-overridden">{{ item.modelName }}</p>
            </template>
          </BaseGrid>`;
-    const { wrapper } = renderBaseGridComponent({ items, columns, template });
+    const { wrapper } = renderBaseGridComponent({ template });
     expect(wrapper.find(getDataTestSelector('default-slot-overridden')).exists()).toBe(true);
   });
 
   it('allows customizing named slots', () => {
     const customItemSlot = `
-      <template #Banner="{ item }">
+      <template #banner="{ item }">
         <p data-test="banner-slot">{{ item.modelName }}</p>
       </template>`;
     const { getDefaultSlot, getScopedSlot } = renderBaseGridComponent({
-      items,
-      columns,
       customItemSlot
     });
 
     expect(getDefaultSlot().exists()).toBe(true);
     expect(getScopedSlot('result').exists()).toBe(false);
     expect(getScopedSlot('banner').exists()).toBe(true);
   });
+
+  it('allows customizing named slots only using kebab case', () => {
+    const customValidItemSlot = `
+      <template #banner="{ item }">
+        <p data-test="banner-slot">{{ item.modelName }}</p>
+      </template>
+      <template #NextQueriesGroup="{ item }">
+        <p data-test="next-queries-group-slot">{{ item.modelName }}</p>
```

Comment:
The data test for this should be `NextQueriesGroup-slot`. 

Below you are using a helper function that just appends `-slot` to the passed model name, and uses to select a DOM element. This means that when you do `getScopedSlot('NextQueriesGroup')` you are running `wrapper.findAll('[data-test="NextQueriesGroup-slot"]'))`, but in your render template there is no element with this data test that could match, therefore the test will always pass, and the assertion worthless.

Additionally, if you just inline the template into the `renderBaseGridComponent` call you would make me very happy. Why create variables that are only going to be used to be passed into a function below? Just inline them in the call, it is much easier.

Reply:
I left :
```html
   <template #NextQueriesGroup="{ item }">
        <p data-test="next-queries-group-slot">{{ item.modelName }}</p>
      </template>
      <template #next-queries-group="{ item }">
        <p data-test="next-queries-group-slot">{{ item.modelName }}</p>
      </template>
```

to demonstrate that an invalid slot is not render and the kebab case it's, it's ok if the `getScopedSlot` doesn't found it.